### ISSUE TITLE
ESLint error for JsonObject

### ISSUE THREAD

**Issue by Andrew-Chen-Wang** [RobinBlomberg/kysely-codegen#181]
version: 0.15.0
```ts
export type Json = JsonValue

export type JsonArray = JsonValue[]

export type JsonObject = {
  [K in string]?: JsonValue
}

export type JsonPrimitive = boolean | number | string | null

export type JsonValue = JsonArray | JsonObject | JsonPrimitive
```

Error: `ESLint: 'K' is defined but never used.(no-unused-vars)`

How do I fix?

ESLint file:

```json
{
  "env": {
    "browser": true,
    "es2021": true
  },
  "extends": ["eslint:recommended", "plugin:react/recommended", "next/core-web-vitals", "prettier"],
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    },
    "ecmaVersion": "latest",
    "sourceType": "module"
  },
  "plugins": ["react"],
  "rules": {
    "indent": ["warn", 2],
    "quotes": ["error", "single"],
    "semi": ["off"]
  }
}

```

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/RobinBlomberg) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/RobinBlomberg/kysely-codegen/issues/181">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/RobinBlomberg/kysely-codegen/issues/181/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/RobinBlomberg/kysely-codegen/issues/181/pledge.svg">
</picture>
</a>

**Comment by RobinBlomberg** [comment]
Hey! I recommend you to exclude the generated file from your ESLint settings, and I very much recommend you to use [@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars/) instead of [no-unused-vars](https://eslint.org/docs/latest/rules/no-unused-vars) as it doesn't work correctly for TypeScript code.

Regardless, to make it easier for future users (and legacy TypeScript users), the next version of kysely-codegen will switch to this equivalent syntax:
```ts
export type JsonObject = {
  [x: string]: JsonValue | undefined;
}
```

Thanks!

**Comment by Andrew-Chen-Wang** [comment]
oh that's great; thanks so much!

**Comment by RobinBlomberg** [comment]
Should be released in [kysely-codegen@0.16.0](https://github.com/RobinBlomberg/kysely-codegen/releases/tag/0.16.0)! 🚀
